packages/vite/src/node/utils.ts

```diff
@@ -561,5 +561,21 @@ export function arraify<T>(target: T | T[]): T[] {
   return Array.isArray(target) ? target : [target]
 }
 
+export function toUpperCaseDriveLetter(pathName: string): string {
+  if (/^\w:/.test(pathName)) {
+    return pathName
+      .split(':')
+      .map((it, index) => {
+        if (index === 0 && it.length === 1) {
+          return it.toUpperCase()
+        } else {
+          return it
+        }
+      })
+      .join(':')
+  }
+  return pathName
+}
```

Comment:
I think we can simplify this function a bit, do you see an issue with a direct replace here?
```suggestion
export function toUpperCaseDriveLetter(pathName: string): string {
  return pathName.replace(/^\w:/, letter => letter.toUpperCase())
}
```